### PR TITLE
Rdf insert

### PR DESCRIPTION
**Pull request by catch-point**
Adds HTTP Basic authentication to rdf-insert component and includes two other components that might be useful with RDF graphs and SPARQL endpoints

### CODE REVIEW

File: graphs/rdf-clear-insert.fbp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# rdf-clear-insert.fbp
+
+# Takes parsed JSON-LD and clears and inserts it into the target-graph (IRI) at the sparql-endpoint.
```
Comment (dbooth-boston) [inline graphs/rdf-clear-insert.fbp]: Okay, but right now we need one that takes parsed RDF (parsed Turtle).  Later we'll need one that takes parsed JSON-LD.

Comment (catch-point) [inline graphs/rdf-clear-insert.fbp]: Do you also want it to clear the remote named graph?

File: graphs/rdf-construct.fbp [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# rdf-construct.fbp
+
+# Retrieves the source-graph (IRI) from the sparql-endpoint and returns parsed JSON-LD
```
Comment (dbooth-boston) [inline graphs/rdf-construct.fbp]: So this (in essence) serializes a graph to JSON-LD and then parses it?

Do we need this?  If so, where?

File: graphs/rdf-clear-insert.fbp (hunk 1 above)
Comment (dbooth-boston) [inline graphs/rdf-clear-insert.fbp]: yes

File: graphs/rdf-construct.fbp (hunk 2 above)
Comment (catch-point) [inline graphs/rdf-construct.fbp]: We need it in rdf-insert-mocha to test round trip. Without it there is no way to ensure rdf-insert succeeded.

Comment (dbooth-boston) [inline graphs/rdf-construct.fbp]: ok

File: graphs/rdf-clear-insert-jsonld.fbp [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# rdf-clear-insert-jsonld.fbp
+
+# Takes parsed JSON-LD and clears and inserts it into the target-graph (IRI) at the sparql-endpoint.
+
+# Exports
+INPORT=request.URL:SPARQL_ENDPOINT
```
Comment (dbooth-boston) [inline graphs/rdf-clear-insert-jsonld.fbp:7]: Suggest SPARQL_ENDPOINT --> SPARQL_SERVER_URI

File: graphs/rdf-clear-insert-jsonld.fbp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+INPORT=regroup.IN:TARGET_GRAPH
```
Comment (dbooth-boston) [inline graphs/rdf-clear-insert-jsonld.fbp]: Suggeste TARGET_GRAPH --> TARGET_GRAPH_URI

File: graphs/rdf-clear-insert-jsonld.fbp [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+INPORT=load.INPUT:JSONLD
```
Comment (dbooth-boston) [inline graphs/rdf-clear-insert-jsonld.fbp]: Suggest JSONLD --> PARSED_JSONLD

File: graphs/rdf-clear-insert-jsonld.fbp [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+INPORT=authFile.KEY:AUTH_FILE_ENV
```
Comment (dbooth-boston) [inline graphs/rdf-clear-insert-jsonld.fbp:10]: A comment further explaining this input would be good

File: graphs/rdf-clear-insert.fbp [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# rdf-clear-insert.fbp
+
+# Takes RDF JS Interface Graph object and clears and inserts it into the target-graph (IRI) at the sparql-endpoint.
+
+# Exports
+INPORT=request.URL:SPARQL_ENDPOINT
```
Comment (dbooth-boston) [inline graphs/rdf-clear-insert.fbp:7]: Suggest SPARQL_ENDPOINT --> SPARQL_SERVER_URI

File: graphs/rdf-clear-insert.fbp [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+INPORT=regroup.IN:TARGET_GRAPH
```
Comment (dbooth-boston) [inline graphs/rdf-clear-insert.fbp]: suggest TARGET_GRAPH --> TARGET_GRAPH_URI

File: graphs/rdf-construct-jsonld.fbp [hunk 9]
```diff
@@ -0,0 +1,32 @@
+# rdf-construct-jsonld.fbp
+
+# Retrieves the source-graph (IRI) from the sparql-endpoint and returns parsed JSON-LD
+
+# Exports
+INPORT=request.URL:SPARQL_ENDPOINT
```
Comment (dbooth-boston) [inline graphs/rdf-construct-jsonld.fbp:7]: suggest SPARQL_ENDPOINT --> SPARQL_SERVER_URI

File: graphs/rdf-construct-jsonld.fbp [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+INPORT=iriGraph.IN:SOURCE_GRAPH
```
Comment (dbooth-boston) [inline graphs/rdf-construct-jsonld.fbp]: suggest SOURCE_GRAPH --> SOURCE_GRAPH_URI

File: graphs/rdf-construct.fbp [hunk 11]
```diff
@@ -0,0 +1,32 @@
+# rdf-construct.fbp
+
+# Retrieves the source-graph (IRI) from the sparql-endpoint and returns an RDF JS Interface Graph object
+
+# Exports
+INPORT=request.URL:SPARQL_ENDPOINT
```
Comment (dbooth-boston) [inline graphs/rdf-construct.fbp:7]: suggest SPARQL_ENDPOINT --> SPARQL_SERVER_URI

File: graphs/rdf-construct.fbp [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+INPORT=iriGraph.IN:SOURCE_GRAPH
```
Comment (dbooth-boston) [inline graphs/rdf-construct.fbp]: suggest add _URI

File: graphs/rdf-insert-jsonld.fbp [hunk 13]
```diff
@@ -0,0 +1,34 @@
+# rdf-insert-jsonld.fbp
+
+# Takes parsed JSON-LD and inserts it into the target-graph (IRI) at the sparql-endpoint.
+
+# Exports
+INPORT=request.URL:SPARQL_ENDPOINT
```
Comment (dbooth-boston) [inline graphs/rdf-insert-jsonld.fbp:7]: suggest SPARQL_ENDPOINT --> SPARQL_SERVER_URI

File: graphs/rdf-insert-jsonld.fbp [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+INPORT=regroup.IN:TARGET_GRAPH
```
Comment (dbooth-boston) [inline graphs/rdf-insert-jsonld.fbp]: suggest add _URI

File: graphs/rdf-insert.fbp [hunk 15]
```diff
@@ -1,27 +1,33 @@
 # rdf-insert.fbp
 
-# Takes parsed JSON-LD and inserts it into the target-graph (IRI) at the sparql-endpoint.
+# Takes RDF JS Interface Graph object and inserts it into the target-graph (IRI) at the sparql-endpoint.
 
 # Exports
 INPORT=request.URL:SPARQL_ENDPOINT
```
Comment (dbooth-boston) [inline graphs/rdf-insert.fbp:7]: suggest SPARQL_ENDPOINT --> SPARQL_SERVER_URI

File: graphs/rdf-insert.fbp [hunk 16]
```diff
@@ -1,27 +1,33 @@
...
 INPORT=regroup.IN:TARGET_GRAPH
-INPORT=load.INPUT:JSONLD
+INPORT=ntriples.INPUT:RDF_GRAPH
```
Comment (dbooth-boston) [inline graphs/rdf-insert.fbp:9]: suggest add _URI

File: graphs/rdf-put-graph.fbp [hunk 17]
```diff
@@ -0,0 +1,49 @@
+# rdf-put-graph.fbp
+
+# Takes parsed JSON-LD and POSTs it to the authority as text/turtle with the target-graph (IRI) as a context parameter.
+
+# Exports
+INPORT=authority.IN:AUTHORITY
+INPORT=repository.IN:REPOSITORY
+INPORT=regroup.IN:TARGET_GRAPH
```
Comment (dbooth-boston) [inline graphs/rdf-put-graph.fbp]: suggest add _URI

File: graphs/rdf-put-graph.fbp [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+INPORT=load.INPUT:JSONLD
```
Comment (dbooth-boston) [inline graphs/rdf-put-graph.fbp]: suggest JSONLD --> PARSED_JSONLD

File: graphs/rdf-insert.fbp (hunk 16 above)
Comment (catch-point) [inline graphs/rdf-insert.fbp:9]: The input here is a parsed/indexed RDF Graph object

File: graphs/rdf-clear-insert-jsonld.fbp (hunk 3 above)
Comment (catch-point) [inline graphs/rdf-clear-insert-jsonld.fbp:7]: The spec uses the term endpoint as in "SPARQL endpoint...The URI at which a SPARQL Protocol service listens for requests from SPARQL Protocol clients.", why do you suggest using the term server-uri?

https://www.w3.org/TR/2013/REC-sparql11-protocol-20130321/#terminology

Comment (dbooth-boston) [inline graphs/rdf-clear-insert-jsonld.fbp:7]: ok, fine to keep it as ENDPOINT then.

**Comment by dbooth-boston** [comment]
LGTM for now, but the test code still seems like it is using a lot of low level noflo calls.  When we can we should enhance our noflo access layer (or common-test layer) to enable higher level test code that does not require as much detailed knowledge of noflo.

**Comment by glenna-m** [comment]
LGTM

**Comment by dbooth-boston** [comment]
LGTM

**Pull request merged**